tox.ini

```diff
@@ -1,23 +1,31 @@
 [tox]
 minversion = 2.4
-envlist = lint,types,py{36,37,38},docs
+envlist = lint,types,py{36,37,38},docs,integration
 
 [testenv]
 deps =
     pretend
-    pytest<6.1 # https://github.com/pytest-dev/pytest-rerunfailures/issues/128
+    pytest
     pytest-cov
     pytest-socket
-    pytest-rerunfailures
+passenv =
+    PYTEST_ADDOPTS
+commands =
+    pytest --ignore-glob '*integration*.py' {posargs:--cov-report term-missing --cov-report html tests}
+
+[testenv:integration]
+deps =
+    {[testenv]deps}
     jaraco.envs
+    munch
     portend
+    pytest-rerunfailures
     pytest-services
-    munch
-    colorama
 passenv =
     PYTEST_ADDOPTS
 commands =
-    pytest {posargs:--cov-report term-missing --cov-report html tests}
+    # Skipping coverage because that should be handled by the other tests
+    pytest -r aR --no-cov tests/test_integration.py
```

Comment:
Looks like this does cause our coverage to drop: https://codecov.io/gh/pypa/twine/pull/704/changes. I still think those cases should be covered by the "fast" tests, and I think including the isolated integration tests in both local and codecov reporting would require some non-trivial configuration.

Reply:
I think something like 80% is a fine code-coverage threshold.